README.md

```diff
@@ -976,6 +976,283 @@ Group membership can be created by:
   $group->addAccount($account);
   ```
 
+### Integrating with Google
+
+Stormpath supports accessing accounts from a number of different 
+locations including Google. Google uses OAuth 2.0 protocol for 
+authentication / authorization and Stormpath can leverage their 
+authorization code (or access tokens) to return an `Account` for 
+a given code.
+
+The steps to enable this functionality into your application include:
+
++ Create a Google Directory
++ Create an `AccountStoreMapping` between a Google Directory and your `Application`
++ Accessing Accounts with Google Authorization Codes or Access Tokens
+
+Google Directories follow behavior similar to mirror directories, but 
+have a `Provider` resource that contains information regarding the Google 
+application that the directory is configured for.
+
+#### Google Provider Resource
+
+A GoogleProvider resource holds specific information needed for working with 
+a Google Directory. It is important to understand the format of the provider 
+resource when creating and updating a Google Directory.
+
+A provider resource can be obtained by accessing the directory’s provider as 
+follows:
+
+```PHP
+$provider = $client->dataStore->getResource("https://api.stormpath.com/v1/directories/1mBDmVgW8JEon4AkoSYjPv/provider",
+    \Stormpath\Stormpath::GOOGLE_PROVIDER);
+```
+
+or, by means of the directory Resource:
+
+```PHP
+$provider = $directory->getProvider();
+```
+
+##### Resource Attributes
+
+* `clientId` : The App ID for your Google application
+* `clientSecret` : The App Secret for your Google application
+* `redirectUri` : The redirection Uri for your Google application
+* `providerId` : The provider ID is the Stormpath ID for the Directory’s account provider
+
+In addition to your application specific attributes, a Provider resource 
+will always contain 3 reserved read-only fields:
+
+* `href` : The fully qualified location of the custom data resource
+* `createdAt` : the UTC timestamp with millisecond precision of when the resource was created in Stormpath as an ISO 8601 formatted string
+* `modifiedAt` : the UTC timestamp with millisecond precision of when the resource was created in Stormpath as an ISO 8601 formatted string
+
+#### Creating a Google Directory
+
+Creating a Google Directory requires that you gather some information 
+beforehand from Google’s Developer Console regarding your application.
+
+* Client ID
+* Client Secret
+* Redirect URI
+
+Creating a Google Directory is very similar to creating a directory within 
+Stormpath. For a Google Directory to be configured correctly, you must 
+specify the correct Provider information.
+
+```PHP
+$provider = $client->dataStore->instantiate(\Stormpath\Stormpath::GOOGLE_PROVIDER);
+$provider->setClientId("857385-m8vk0fn2r7jmjo.apps.googleusercontent.com");
+$provider->setClientSecret("ehs7_-bA7OWQSQ4");
+$provider->setRedirectUri("https://myapplication.com/authenticate");
```

Comment:
Instead of providing the instructions with the setters, please provide the instructions where we use the magic methods. For example:

```php
$provider->clientId = "857385-m8vk0fn2r7jmjo.apps.googleusercontent.com";
```